cpmpy/transformations/linearize.py

```diff
@@ -277,6 +302,17 @@ def linearize_constraint(lst_of_expr, supported={"sum","wsum"}, reified=False):
             [cpm_expr] = canonical_comparison([cpm_expr])  # just transforms the constraint, not introducing new ones
             lhs, rhs = cpm_expr.args
 
+            # check trivially true/false (not allowed by PySAT Card/PB)
+            if cpm_expr.name in ('<', '<=', '>', '>=') and is_num(rhs):
+                lb,ub = lhs.get_bounds()
+                t_lb = eval_comparison(cpm_expr.name, lb, rhs)
+                t_ub = eval_comparison(cpm_expr.name, ub, rhs)
+                if t_lb and t_ub:
+                    continue
+                elif not t_lb and not t_ub:
+                    newlist += linearize_constraint([BoolVal(False)], supported=supported) # post the linear version of False
```

Comment:
that seems like a very indirect way... you are in this function; just do the 'or' check here, then return Or(false), otherwise sum(...)?

Reply:
If I move that code here, then linearize can't handle the expression of BoolVal(False) (it's a trivial expression, but it still solver-dependent). So this seems like the right way to me.

Reply:
I meant just copy the 2-line if here? I guess this is such a rare case that it doesn't actually matter to save 1 function call, so you can also leave it as is I realize now

Reply:
Ah duplicate it; it is only 2 lines, I agree, but this can change in the future, and I rather stick to the approach where we just use linearize to linearize something not linear